lib/connection.ts

```diff
@@ -314,6 +343,14 @@ export class Connection extends Entity {
         log.connection("[%s] Trying to create a new amqp connection.", this.id);
         this._connection.connect();
         this.actionInitiated++;
+
+        if (abortSignal) {
```

Comment:
There are a lot more common patterns repeated in this project like timeouts, removing of listeners etc. I wanted to refrain from introducing too much of a churn, so chose to repeat the code rather than introduce modularity

Reply:
Is there any reason we can't take the _current_ duplication we're adding and consolidating it? I get what you're saying but there's three instances in this one PR alone.

Reply:
Fair point.

After using optional chaining, the above reduces to
```js
if (abortSignal?.aborted) {
     onAbort();
} else {
     abortSignal?.addEventListener("abort", onAbort);
}
```

At this point, keeping it inline makes it very clear as to what is being done. Moving to a function hides away the intent and adds 1 hop to understanding what we do. I would prefer making a function if the code being moved is complex or long. Otherwise, simplicity of keeping it inline and understanding at a quick glance what is happening wins for me.

Reply:
I think it's more the general pattern that seems to be all over. 

In Service Bus (different project!) I added some helpers for that. We can leave that for out-of-scope for this PR but I do worry about this kind of duplication.